frame/contracts/src/wasm/env_def/macros.rs

```diff
@@ -109,14 +109,20 @@ macro_rules! unmarshall_then_body_then_marshall {
 		>(|| {
```

Comment:
If I'm correct this macro arm is for when it returns `ReturnCode`, (or at least there is no currently other usage of it).

Why not also automatically run `err_into_return_code` here ?

Reply:
Because `err_into_return_code` is called onto a `DispatchError` and returns a `Result`. But at that point in the macro it must have already decided whether we are `Ok` or `Err`.

Reply:
I just wonder why shouldn't we decide here ?
Like we have a `Result<ReturnCode, TrapReason>`, we can automatically change the `Err(TrapReason::SupervisorError(CodeNotFound))` to `Ok(ReturnCode::CodeNotFound)` at this moment.

and then seal_transfer can be simply written:
```rust
	seal_transfer(
		...
	) -> ReturnCode => {
		...

		ctx.ext.transfer(&callee, value)?;
		Ok(ReturnCode::Success)
	},
```

But ok it might add more confusion than help. But it could allow not to miss some `err_into_return_code` to be called I feel.

Anyway fine

Reply:
Okay I see. Not sure how this play out with `exec_into_return_code`. Any ideas there?

Reply:
I guess `exec_into_return_code` would just need to convert `ExecResult::Err:{origin: Callee}` to `Ok(ReturnCode::CalleeTrapped)` (and other err to err and other ok to ok).

But if it is confusing we can merge as it is.

Reply:
I would rather merge is at is and have these further improvements in another PR to not completely shave the yak here. Maybe you can even put up a PR with your suggestions? I will gladly merge it.

Reply:
Yes ok, I'll make a try